test/pkg/broker/broker.go

```diff
@@ -70,6 +76,10 @@ func CreatorWithOptions(client *eventingtestlib.Client, version string, brokerOp
 		panic(fmt.Sprintf("error getting KafkaBroker class from env '%v'", err))
 	}
 
+	return CreatorForClassWithOptions(class, client, version, brokerOptions, configOptions)
+}
+
+func CreatorForClassWithOptions(class string, client *eventingtestlib.Client, version string, brokerOptions []resources.BrokerOption, configOptions []ConfigOptions) string {
```

Comment:
We can reduce the class parameter to a BrokerOption

Reply:
There could even be a function already with that purpose.

Reply:
I thought about it, but I didn't want to do it. It is because, if I make the class a BrokerOption, then I need to create a separate function that reads the env var for class anyway. So, the number of functions won't change and we would have function that's very implicit about the class.

Reply:
It's the parameters of this function that I don't think it's readable and maintainable long term, not the functions number, so I still think it's worth doing it now.

I had to deal with this bad pattern in the past for reconciler tests due to us passing raw inputs to function instead of proper arguments and objects and I wasted so much time refactoring code due to the quick and dirty additions of parameters without having a good arguments design.

This is exploding in parameters, like class (and, fwiw existing version) can and should be `BrokerOption`s.